src/commands/edge-check.test.js

```diff
@@ -0,0 +1,71 @@
+const should = require(`should`);
+const sinon = require(`sinon`);
+require(`should-sinon`);
+
+const util = require(`./../util`);
+const testSubject = require(`./edge-check`);
+
+// Scope vars that we’ll redefine on each test.
+let options, edgeCheckStub, Fastly;
+
+// Begin testing.
+describe(`edge-check`, () => {
+  // Before and after each test, we reset some stuff
+  // to ensure a fresh set of arguments. See .beforeEach()
+  // and .afterEach() near the bottom of this file.
+
+  it(`should stop without an API key`, (done) => {
+    delete options.apikey;
+    testSubject(options, Fastly, util);
+    util.apiKeyPresent.should.be.calledOnce();
+    Fastly.should.not.be.called();
+    done();
+  });
+
+  it(`should instantiate Fastly with API key`, (done) => {
+    testSubject(options, Fastly, util);
+    Fastly.should.be.calledOnce();
+    Fastly.getCalls()[0].args.should.deepEqual([options.apikey]);
+    done();
+  });
+
+  it(`should instantiate ResponseHandler`, (done) => {
+    testSubject(options, Fastly, util);
+    util.ResponseHandler.should.be.calledOnce();
+    done();
+  });
+
+  it(`should invoke .edgeCheck()`, (done) => {
+    testSubject(options, Fastly, util);
+    edgeCheckStub.should.be.calledOnce();
```

Comment:
Let’s add an assertion below this that verifies `edgeCheckStub()`’s first argument is `url`, since it’s important it gets passed to `fastly`.